programs/uxd/src/instructions/credix_lp/rebalance_request_execute_from_credix_lp_depository.rs

```diff
@@ -0,0 +1,646 @@
+        CREDIX_LP_DEPOSITORY_NAMESPACE,
+        credix_global_market_state.as_ref(),
+        collateral_mint.as_ref(),
+        &[ctx.accounts.depository.load()?.bump],
+    ]];
+
+    // ---------------------------------------------------------------------
+    // -- Phase 1
+    // -- Check if the withdraw redeem period is active at the moment
+    // ---------------------------------------------------------------------
+
+    let start_of_redeem_phase_timestamp = ctx
+        .accounts
+        .credix_withdraw_epoch
+        .go_live
+        .checked_add(ctx.accounts.credix_withdraw_epoch.redeem_seconds.into())
+        .ok_or(UxdError::MathError)?;
+    let end_of_redeem_phase_timestamp = start_of_redeem_phase_timestamp
+        .checked_add(ctx.accounts.credix_withdraw_epoch.redeem_seconds.into())
+        .ok_or(UxdError::MathError)?;
+    require!(
+        (start_of_redeem_phase_timestamp..end_of_redeem_phase_timestamp)
+            .contains(&Clock::get()?.unix_timestamp),
+        UxdError::InvalidCredixWithdrawEpochRequestPeriod,
+    );
+
+    // ---------------------------------------------------------------------
+    // -- Phase 2
+    // -- Fetch all current onchain state
+    // ---------------------------------------------------------------------
+
+    let credix_lp_depository_collateral_amount_before: u64 =
+        ctx.accounts.depository_collateral.amount;
+    let identity_depository_collateral_amount_before: u64 =
+        ctx.accounts.identity_depository_collateral.amount;
+    let profits_beneficiary_collateral_amount_before: u64 =
+        ctx.accounts.profits_beneficiary_collateral.amount;
+
+    let liquidity_collateral_amount_before: u64 = ctx.accounts.credix_liquidity_collateral.amount;
+    let outstanding_collateral_amount_before: u64 = ctx
+        .accounts
+        .credix_global_market_state
+        .pool_outstanding_credit;
+
+    let total_shares_supply_before: u64 = ctx.accounts.credix_shares_mint.supply;
+    let total_shares_value_before: u64 = liquidity_collateral_amount_before
+        .checked_add(outstanding_collateral_amount_before)
+        .ok_or(UxdError::MathError)?;
+
+    let owned_shares_amount_before: u64 = ctx.accounts.depository_shares.amount;
+    let owned_shares_value_before: u64 = compute_value_for_shares_amount_floor(
+        owned_shares_amount_before,
+        total_shares_supply_before,
+        total_shares_value_before,
+    )?;
+
+    // ---------------------------------------------------------------------
+    // -- Phase 3
+    // -- Compute the profits and the overflow of the depository we will want to withdraw
+    // ---------------------------------------------------------------------
+
+    let redeemable_amount_under_management = checked_convert_u128_to_u64(
+        ctx.accounts
+            .depository
+            .load()?
+            .redeemable_amount_under_management,
+    )?;
+
+    let profits_collateral_amount = owned_shares_value_before
```

Comment:
Here we are recalculating overflow + profit at credix withdraw execution time, which is Xd after the request is made.

Here are two choices:
- Withdraw the exact amount of what have been requested
- Update targets at withdraw time and either withdraw 100% of what have been asked, or <100%.

Note, between the withdraw request time and the execute request time, following things have happened:
- Profits increased
- Possibility of `overflow_value` to be bigger or lower
    - `overflow_value` is bigger now if the UXD circulating supply decreased.
    - `overflow_value` is lower if the UXD circulating supply increased.

Reply:
Issue I see with updating target is that you may never been able to collect profits as `overflow_value` may increase and grab all available collateral.

Reply:
Yes everything you said is correct. I think we are on the same page.

First of all I think this issue is not a very big problem in practice as the time between request/execute is small and it's a self-healing process as the next month's request will do all the necessary adjustment automatically.

That being said there is something we could do better:
- We could request an "overshot" amount during the create part of the withdraw request
- Then we could execute the exact amount during the execute part of the withdraw request

This would effectively tell credix everymonth "we want to withdraw everything" then we would "actually" just withdraw only what we need and the rest would return to the liquid pool after that if anything we haven't withdrawn

Reply:
I'll make that change now, I don't see why not. That's a good catch! thanks!

Keeping this thread open, let me know if my answer make sense and the proposed solution is sufficient in your opinion

Reply:
Discussed more in depth offline, keeping the original maths of just requesting the amount we "think" we are going to be withdrawing during creation phase